HousingSearchApi/V1/Infrastructure/Filtering/FilterFactory.cs

```diff
@@ -0,0 +1,32 @@
+using Hackney.Shared.HousingSearch.Gateways.Models.Transactions;
+using HousingSearchApi.V1.Boundary.Requests;
+using HousingSearchApi.V1.Interfaces.Filtering;
+using Nest;
+
+namespace HousingSearchApi.V1.Infrastructure.Filtering
+{
+    public class FilterFactory : IFilterFactory
+    {
+        /// <summary>
+        /// Method to perform filtering from start to end dates
+        /// </summary>
+        public QueryContainer Filter<T, TRequest>(TRequest request, QueryContainerDescriptor<T> q) where T : class where TRequest : class
+        {
+            // ToDo: implement factory for choosing type of filter
```

Comment:
is there a plan to do this?

Reply:
Yes, I will add this a little bit later, on the next week